augur/io/metadata.py

```diff
@@ -158,12 +167,11 @@ def read_table_to_dict(table, duplicate_reporting=DataErrorMethod.ERROR_FIRST, i
         try:
             # Note: this sort of duplicates _get_delimiter(), but it's easier if
             # this is separate since it handles non-seekable buffers.
-            dialect = csv.Sniffer().sniff(table_sample, VALID_DELIMITERS)
-        except csv.Error as err:
-            raise AugurError(
-                f"Could not determine the delimiter of {table!r}. "
-                "File must be a CSV or TSV."
-            ) from err
+            dialect = csv.Sniffer().sniff(table_sample, delimiters)
+        except csv.Error as error:
+            # This assumes all csv.Errors imply a delimiter issue. That might
+            # change in a future Python version.
+            raise InvalidDelimiter from error
```

Comment:
_non-blocking_

Oh, huh. TIL you can `raise ClassName` and it'll be equivalent to `raise ClassName()` (i.e. raises an instance). :-)